### ISSUE TITLE
Improperly configured FacePunch.Steamworks.Posix Unity plugin

### ISSUE THREAD

**Issue by alkalinethunder** [Facepunch/Facepunch.Steamworks#645]
It seems that, although Facepunch.Steamworks lists itself as supporting Linux natively (awesome!), the Unity plugin file for POSIX systems (macOS/Linux) is misconfigured.

After launching the Unity Editor for one of my gamedev projects that uses this library, I noticed that several hundred script compiler errors were preventing me from launching the full editor. Upon reading the logs I found that they were all errors related to missing Steamworks namespaces and undeclared identifiers.

A quick Google search brought up this [Unity Answers thread](https://answers.unity.com/questions/1778267/cannot-use-facepunch-steamworks-library.html) describing the exact same issue.

The suggested fix was to open the `Assets/Plugins/Facepunch.Steamworks/Facepunch.Steamworks.Posix` plugin and  change the "OS" setting under "Platform Configuration" from "OSX" to "Linux."

Doing the above on my project fixed the issue. But perhaps this is something that could be configured by default in future releases? Seems like a good idea to split the Posix plugin into a separate OSX and Linux plugin since Unity seems to only let you select either, not both, for a given plugin.

**Comment by rgarlik** [comment]
This happened to me too when opening my Unity project on a Linux machine. I'm not really sure why macOS and Linux are in one plugin, seems like an odd design decision.

**Comment by DasProfGames** [comment]
Hello,
I have the same issue.
But in my case I cannot edit the Facepunch.Steamworks.Posix to change macOS to Linux.
Is it because I have install it via PAckage Manager and that the files are in Package folder (and not in Assets) ?
Thanks in advance for your reply.

**Comment by alkalinethunder** [comment]
Glad to know I wasn't the only one having the issue.

I almost wonder if the simplest fix would be to just duplicate the Posix plugin, and then have one for macOS and one for Linux.

**Comment by khyperia** [comment]
I've submitted a PR that should fix this, https://github.com/Facepunch/Facepunch.Steamworks/pull/789, using the method described in https://github.com/Facepunch/Facepunch.Steamworks/issues/665#issuecomment-1324475537.

tl;dr set the OS field in the editor settings for the dll to "Any OS" and add a Define Constraint of `UNITY_EDITOR_OSX || UNITY_EDITOR_LINUX`

![Image](https://github.com/user-attachments/assets/5868945b-1993-48d7-b063-4df9e0621e9c)
